### PR TITLE
Upgrade versions

### PR DESCRIPTION
**Pull request by johncblandii**
## Breaking Changes

This module includes breaking changes due to upgrading from [terraform-aws-security-group](https://github.com/cloudposse/terraform-aws-security-group) v1 to v2. You can read the full details of the security group changes and how to migrate in migration notes linked below under "references", but the short story is this:

- If you were using this modules default value of `true` for `security_group_create_before_destroy` then you need not make any changes. If you were explicitly setting it to `false`, we strongly advise you to read the migration notes because that setting previously did not work, raising the question of whether you want to pay the price of converting to the new module with working `false` behavior or perform the recommended upgrade to the `true`.
- If you are referring by ID to the security group created by this module in other security group's rules outside the Terraform plan that controls this one, then you should read the security group migration notes discussion of the new input `preserve_security_group_id` and probably set it to `true`

## what
* Upgrade versions to the latest

### 

#### Key changes in `terraform-aws-security-group` v2.0 affecting this release

* create_before_destory default changed from false to true
* preserve_security_group_id added, defaults to false
* Terraform version 1.0.0 or later required

## why
* The module fails when used with the latest https://github.com/cloudposse/terraform-aws-components

## references
* https://github.com/cloudposse/terraform-aws-components/pull/602
* Migration Notes for [Security Group v2.0](https://github.com/cloudposse/terraform-aws-security-group/blob/master/docs/migration-v1-v2.md)

### CODE REVIEW

**Review by Benbentwo — APPROVED** [review]

**Comment by johncblandii** [comment]
> @johncblandii please run these commands and commit the changes
> 
> ```
> make init
> make github/init
> make readme
> ```

Done

File: sg.tf [hunk 1]
```diff
@@ -53,7 +53,7 @@ locals {
 
 module "security_group" {
   source  = "cloudposse/security-group/aws"
-  version = "1.0.1"
+  version = "2.0.0"
```
Comment (Nuru) [inline sg.tf:56]: This module should wait until https://github.com/cloudposse/terraform-aws-security-group/issues/43 is fixed and then use the fixed module.

```suggestion
  version = "2.0.1"
```

[the remark this answers is not in the document]
Comment (Nuru) [reply sg.tf:56]: @johncblandii I'm sorry, I gave you a big and vague task without nearly enough direction. Please forgive me.

I added the relevant documentation to the PR description. However, you still need to add the [`preserve_security_group_id`](https://github.com/cloudposse/terraform-aws-security-group/blob/a1b4b2dc28dc17c1b597b580fd867a069e0a2844/exports/security-group-variables.tf#L115-L129) variable to the `security-group-variables.tf` file and pass it through to the security group module.

Comment (johncblandii) [reply sg.tf:56]: Any ETAs on when that will be completed?

**Review by Nuru — APPROVED** [review]

**Pull request merged**